### ISSUE TITLE
request to add query optimization to get_collection

### ISSUE THREAD

**Issue by zabano** [ColtonProvias/sqlalchemy-jsonapi#17]
Hi Colton,

Thank you for developing this great package. I am very grateful.
Would it be possible to add some query optimization to the get_collection method based on the included resources? for example, i added the following code to line 621 of [serializer.py](https://github.com/ColtonProvias/sqlalchemy-jsonapi/blob/master/sqlalchemy_jsonapi/serializer.py#L621).

```
for local in include.keys():
  if local:
      loader = joinedload(local)
      for remote in include[local]:
          if '.' in remote:
              for r in remote.split('.'):
                  loader.joinedload(r)
          else:
              loader.joinedload(remote)
      collection = collection.options(loader)
```

Thanks,
Omar

**Comment by ColtonProvias** [comment]
Definitely!  I'm a fan of optimization.  I noticed one fork of this project even has caching within `get_collection` and I want to implement that as well.

(Note to self on location of joinedload in SQLAlchemy: http://docs.sqlalchemy.org/en/latest/orm/loading_relationships.html#sqlalchemy.orm.joinedload)
